rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -159,6 +173,48 @@ fn wireless_config_to_dbus(conn: &WirelessConnection) -> NestedHash {
     ])
 }
 
+/// Converts a MatchConfig struct into a HashMap that can be sent over D-Bus.
+///
+/// * `match_config`: MatchConfig to convert.
+fn match_config_to_dbus(match_config: &MatchConfig) -> HashMap<&str, zvariant::Value> {
+    let mut match_config_dbus = HashMap::new();
+
+    let drivers: Value = match_config
+        .driver
+        .iter()
+        .map(|dr| dr.to_string())
+        .collect::<Vec<String>>()
```

Comment:
note: No need to specify the type of the element. However, I am fine with keeping it.

```suggestion
        .collect::<Vec<_>>()
```